services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy

```diff
@@ -45,4 +51,100 @@ class SwitchesSpec extends BaseSpecification {
         e.statusCode == HttpStatus.NOT_FOUND
         e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID not found."
     }
+
+    def "Systems allows to get a flow that goes through a switch"() {
+        given: "Two active not neighboring switches with two diverse paths at least"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.paths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() >= 3
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "A protected flow"
+        def protectedFlow = flowHelper.randomFlow(switchPair)
+        protectedFlow.allocateProtectedPath = true
+        flowHelper.addFlow(protectedFlow)
+
+        and: "A single switch flow"
+        def simpleFlow = flowHelper.singleSwitchFlow(switchPair.src)
+        flowHelper.addFlow(simpleFlow)
+
+        when: "Get all flows going through the involved switches"
+        def flowPathInfo = northbound.getFlowPath(protectedFlow.id)
+        def mainPath = pathHelper.convert(flowPathInfo)
+        def protectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+
+        def mainSwitches = pathHelper.getInvolvedSwitches(mainPath)*.dpId
+        def protectedSwitches = pathHelper.getInvolvedSwitches(mainPath)*.dpId
+        def involvedSwitchIds = (mainSwitches + protectedSwitches).unique()
+
+        then: "The created flows are in the response list from the src switch"
+        def switchFlowsResponseSrcSwitch = northbound.getSwitchFlows(switchPair.src.dpId)
+        switchFlowsResponseSrcSwitch.size() == 2
+        switchFlowsResponseSrcSwitch*.id.sort() == [protectedFlow.id, simpleFlow.id].sort()
+
+        and: "The protectedFlow only is in the response list from the involved switch(except the src switch)"
+        involvedSwitchIds.findAll { it != switchPair.src.dpId }.each { switchId ->
+            def getSwitchFlowsResponse = northbound.getSwitchFlows(switchId)
+            getSwitchFlowsResponse.size() == 1
+            getSwitchFlowsResponse[0].id == protectedFlow.id
+        }
+
+        when: "Get all flows going through the src switch based on the port of the main path"
+        def getSwitchFlowsResponse1 = northbound.getSwitchFlows(switchPair.src.dpId, mainPath[0].portNo)
+
+        then: "The protected flow only is in the response list"
+        getSwitchFlowsResponse1.size() == 1
+        getSwitchFlowsResponse1[0].id == protectedFlow.id
+
+        when: "Get all flows going through the src switch based on the port of the protected path"
+        def getSwitchFlowsResponse2 = northbound.getSwitchFlows(switchPair.src.dpId, protectedPath[0].portNo)
+
+        then: "The protected flow only is in the response list"
+        getSwitchFlowsResponse2.size() == 1
+        getSwitchFlowsResponse2[0].id == protectedFlow.id
+
```

Comment:
Covered ISL ports on src. Would like to see more:
1. ISL port on intermediate switch
2. Flow port on src or dst for protected flow (only covered for single-switch now)